nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutTCP.java

```diff
@@ -0,0 +1,270 @@
+        }
+
+        try {
+            String delimiter = getDelimiter(context, flowFile);
+            ByteArrayOutputStream content = readContent(session, flowFile);
+            if (delimiter != null) {
+                content = appendDelimiter(content, delimiter);
+            }
+            sender.send(content.toByteArray());
+            session.transfer(flowFile, REL_SUCCESS);
+            session.commit();
+        } catch (Exception e) {
+            getLogger().error("Exception while handling a process session, transferring {} to failure.", new Object[] { flowFile }, e);
+            onFailure(context, session, flowFile);
+        } finally {
+            // If we are going to use this sender again, then relinquish it back to the pool.
+            if (!isConnectionPerFlowFile(context)) {
+                relinquishSender(sender);
+            } else {
+                sender.close();
+            }
+        }
+    }
+
+    /**
+     * event handler method to perform the required actions when a failure has occurred. The FlowFile is penalized, forwarded to the failure relationship and the context is yielded.
+     *
+     * @param context
+     *            - the current process context.
+     *
+     * @param session
+     *            - the current process session.
+     * @param flowFile
+     *            - the FlowFile that has failed to have been processed.
+     */
+    protected void onFailure(final ProcessContext context, final ProcessSession session, final FlowFile flowFile) {
+        session.transfer(session.penalize(flowFile), REL_FAILURE);
+        session.commit();
+        context.yield();
+    }
+
+    /**
+     * Helper method to read the FlowFile content stream into a ByteArrayOutputStream object.
+     *
+     * @param session
+     *            - the current process session.
+     * @param flowFile
+     *            - the FlowFile to read the content from.
+     *
+     * @return ByteArrayOutputStream object containing the FlowFile content.
+     */
+    protected ByteArrayOutputStream readContent(final ProcessSession session, final FlowFile flowFile) {
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream((int) flowFile.getSize() + 1);
+        session.read(flowFile, new InputStreamCallback() {
+            @Override
+            public void process(final InputStream in) throws IOException {
+                StreamUtils.copy(in, baos);
+            }
+        });
+
+        return baos;
+    }
+
+    /**
+     * Helper method to append a delimiter to the message contents.
+     * 
+     * @param content
+     *            - the message contents.
+     * @param delimiter
+     *            - the delimiter value.
+     *
+     * @return ByteArrayOutputStream object containing the new message contents.
+     */
+    protected ByteArrayOutputStream appendDelimiter(final ByteArrayOutputStream content, final String delimiter) {
+        content.write(delimiter.getBytes(), 0, delimiter.length());
```

Comment:
I think Joe mentioned this already, but we should use a charset with getting the bytes